### ISSUE TITLE
Add React 18 support

### ISSUE THREAD

**Issue by tpacent** [microsoft/fluentui-system-icons#486]
For some reason React [is constrained to](https://github.com/microsoft/fluentui-system-icons/blob/main/packages/react-icons/package.json#L55) `"react": ">=16.8.0 <18.0.0"` in `peerDependencies`.  
When running `npm install`, it complains (see below) the dependency cannot be satisfied unless I use `--legacy-peer-deps`.

Could you alter the upper constraint boundary to include react 18?

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: data@1.0.0
npm ERR! Found: react@18.2.0
npm ERR! node_modules/react
npm ERR!   react@"^18.2.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@">=16.8.0 <18.0.0" from @fluentui/react-icons@2.0.179
npm ERR! node_modules/@fluentui/react-icons
npm ERR!   @fluentui/react-icons@"*" from the root project
```

**Comment by NWessel** [comment]
Yes please :)

**Comment by tomi-msft** [comment]
This has been fixed by this pr: #517
